RFC/src/RFC-0110_BaseNodes.md

```diff
@@ -1 +1,185 @@
+clients, block explorers, wallets, and Tari applications:
+
+  * Block queries
+  * Kernel data queries
+  * Transaction queries
+  * Submission of new transactions
+
+### Transaction validation and propagation
+
+Base nodes can notified of new transactions from
+* connected peers
+* clients via APIs
+
+When a new transaction has been received, it has the `unvalidated` [ValidationState]. The transaction is then passed
+to the transaction validation service, where its state will become one of `rejected` or `validated`.
+
+The transaction validation service checks that:
+
+* all inputs to the transaction are valid [UTXO]s
+* all inputs are not duplicated
+* all inputs are able to be spent (they're not time-locked)
+* all inputs are signed by their owners
+* all outputs have valid [range proof]s
+* all outputs don't already exist in the [UTXO] set.
+* the transaction excess has a valid signature
+* the transaction excess is a valid public key (this proves that $$\Sigma (\mathrm{inputs} - \mathrm{outputs} - \mathrm(fees) = 0$$)
+
+`Rejected` transactions are dropped silently.
+
+`Validated` transactions are
+* Added to the [mempool]
+* forwarded to peers using the transaction [BroadcastStrategy].
+
+### Block validation and propagation
+
+The Block validation and propagation process is analogous to that of transactions.
+
+New blocks are received from the peer-to-peer network, for from an API if the Base Node is connected to a Miner.
+
+When a new block is received, is is assigned the `unvalidated` [ValidationState]. The block is then passed to the
+block validation service. The validation service checks that
+
+* the block hasn't been processed before
+* every [transaction] in the block is valid,
+* the proof-of-work is valid
+* the block header is well-formed
+* the block is being added to the chain with the highest accumulated proof-of-work.
+  * it is possible for the chain to temporarily fork; base nodes SHOULD account for forks up to some configured depth
+  * it is possible that blocks may be received out of order; particularly while syncing. Base Nodes SHOULD keep blocks
+    that have block heights greater than the current chain tip in memory for some preconfigured period.
+* the sum of all excesses is a valid public key (this proves that $$\Sigma (\mathrm{inputs} - \mathrm{outputs} - \mathrm(fees) = 0$$)
+
+`Rejected` blocks are dropped silently. Base Nodes MAY ban peers that consistently pass on bad blocks and add them to a
+blacklist.
+
+`Validated` blocks are
+* added to the [blockchain]
+* forwarded to peers using the block [BroadcastStrategy].
+
+In addition, when a block has been validated and added to the blockchain:
+* The mempool MUST also remove all transactions that are present in the newly validated block.
+* The UTXO set MUST be updated; removing all inputs in the block, and adding all the new outputs in it.
+
+### Seeding nodes and Synchronizing the chain
```

Comment:
Everywhere else you use the british spelling of synchronising but here you use the z.

Personally I am delighted we can use the correct spelling in our repo.

Reply:
argh. How did that slip though?

Reply:
FYP: 

> Everywhere else you use the <s>british</s> correct spelling of synchronising but here you use the z.